contracts/nexus-gateway/src/contract/execute.rs

```diff
@@ -86,23 +26,61 @@ pub fn route_to_router(
     Ok(Response::new().add_messages(router.route(msgs)))
 }
 
-pub fn route_to_nexus(
+pub fn route_message_with_token_to_nexus(
     storage: &mut dyn Storage,
+    querier: QuerierWrapper,
+    info: MessageInfo,
+    msg: router_api::Message,
+) -> Result<Response<nexus::execute::Message>> {
+    let token = match info.funds.as_slice() {
+        [token] => token.clone(),
+        _ => bail!(Error::InvalidToken(info.funds)),
+    };
+
+    route_message_to_nexus(storage, querier, msg, Some(token))?
```

Comment:
Create the BankMsg here and then call `route_message_to_nexus` and chain the response to simplify `route_message_to_nexus` and `route_messages_to_nexus`

Reply:
Not a bad idea.

Reply:
`route_message_to_nexus` has to take the token in cuz the nexus message itself needs it. In that case, I don't think the proposal makes it cleaner.

Reply:
replied [here](https://github.com/axelarnetwork/axelar-amplifier/pull/624#discussion_r1761970028)